Build error: jaxlib 0.7.0, GCC 13.2.0

### Description

I'm seeing the following build errors:
```
ERROR: /tmp/root/spack-stage/spack-stage-py-jaxlib-0.7.0-pynifaoelsg7wplni3lijorw3jtxkhsl/spack-src/jaxlib/mosaic/dialect/gpu/BUILD:80:11: Compiling jaxlib/mosaic/dialect/gpu/mosaic_gpu.cc failed: (Exit 1): cc failed: error executing CppCompile command (from target //jaxlib/mosaic/dialect/gpu:mosaic_gpu)
...
jaxlib/mosaic/dialect/gpu/mosaic_gpu.cc:536:30: error: 'cast' was not declared in this scope
  536 |   auto custom_primitive_op = cast<CustomPrimitiveOp>((*this)->getParentOp());
      |                              ^~~~
...
jaxlib/mosaic/dialect/gpu/mosaic_gpu.cc:536:52: error: expected primary-expression before '>' token
  536 |   auto custom_primitive_op = cast<CustomPrimitiveOp>((*this)->getParentOp());
      |                                                    ^
```
I'm guessing this has something to do with newer C++ standards not being followed. Can we get a quick patch for this?

### System info (python version, jaxlib version, accelerator, etc.)

* Python 3.13.5
* jaxlib 0.7.0
* GCC 13.2.0

* [build log](https://github.com/user-attachments/files/21389605/spack-build-out.txt.gz)
* [build env](https://github.com/user-attachments/files/21389604/spack-build-env.txt.gz)

--- a/jaxlib/mlir/_mlir_libs/jax_mlir_ext.cc
+++ b/jaxlib/mlir/_mlir_libs/jax_mlir_ext.cc
@@ -69,8 +69,9 @@ namespace {
 // Returns true if a location is a NameLoc with a FileLineColLoc child. We
 // assume the NameLoc names a function name in a frame in this case.
 bool IsFrameNameLocation(mlir::Location location) {
-  return isa<mlir::NameLoc>(location) &&
-         isa<mlir::FileLineColLoc>(cast<mlir::NameLoc>(location).getChildLoc());
+  return mlir::isa<mlir::NameLoc>(location) &&
+         mlir::isa<mlir::FileLineColLoc>(
+             mlir::cast<mlir::NameLoc>(location).getChildLoc());
 }
 
 // Split a location into an operation type and an operation name, and a tail
@@ -87,7 +88,7 @@ void ParseLocation(mlir::Location& location, llvm::StringRef& op_type,
     } else {
       op_name = name;
     }
-    location = cast<mlir::NameLoc>(location).getChildLoc();
+    location = mlir::cast<mlir::NameLoc>(location).getChildLoc();
   }
 }
 

--- a/jaxlib/mosaic/dialect/gpu/mosaic_gpu.cc
+++ b/jaxlib/mosaic/dialect/gpu/mosaic_gpu.cc
@@ -522,7 +522,8 @@ llvm::LogicalResult BroadcastInDimOp::verify() {
 }
 
 llvm::LogicalResult ReturnOp::verify() {
-  auto custom_primitive_op = cast<CustomPrimitiveOp>((*this)->getParentOp());
+  auto custom_primitive_op =
+      mlir::cast<CustomPrimitiveOp>((*this)->getParentOp());
 
   // The operand number and types must match the custom primitive signature.
   const auto& results = custom_primitive_op->getResultTypes();

